src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1816,7 +1816,19 @@ internal sealed override bool HasUnscopedRefAttribute
 
         internal sealed override bool UseUpdatedEscapeRules => ContainingModule.UseUpdatedEscapeRules;
 
-        internal sealed override bool IsCallerUnsafe => ContainingModule.UseUpdatedMemorySafetyRules && IsDeclaredRequiresUnsafe;
+        internal sealed override CallerUnsafeMode CallerUnsafeMode
+        {
+            get
+            {
+                if (ContainingModule.UseUpdatedMemorySafetyRules)
+                {
+                    return IsDeclaredRequiresUnsafe ? CallerUnsafeMode.Explicit : CallerUnsafeMode.None;
+                }
+
+                return this.HasParameterContainingPointerType() || ReturnType.ContainsPointerOrFunctionPointer()
```

Comment:
Can a point appear elsewhere in the signature? For instance, `where T : I<int*[]>`
Comment also applies to PEPropertySymbol: `E.extension<T>(T).Property` (with constraint like above) or `E.extension(int*[]).Property` may be scenarios

Reply:
I'm not sure that should be considered caller-unsafe in the compat mode, because calling such method wouldn't be considered unsafe previously either. Similarly we were discussing extern unsafe rule (I will add you to the offline discussion.) @333fred thoughts?